include/zephyr/rtio/rtio_spsc.h

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#ifndef ZEPHYR_RTIO_SPSC_H_
+#define ZEPHYR_RTIO_SPSC_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <zephyr/sys/atomic.h>
+
+/**
+ * @brief RTIO SPSC API
+ * @defgroup rtio_spsc RTIO SPSC API
+ * @ingroup rtio
+ * @{
+ */
+
+/**
+ * @file rtio_spsc.h
+ *
+ * @brief A lock-free and type safe power of 2 fixed sized SPSC Queue using a
+ * ringbuffer and atomics to ensure coherency.
+ *
+ * This SPSC implementation works on an array which wraps using a power of two
+ * size and uses a bit mask to perform a modulus. Atomics are used to allow
+ * single-producer single-consumer safe semantics without locks. Elements are
+ * expected to be of a fixed size. The API is type safe as the underlying buffer
+ * is typed and all usage is done through macros.
+ *
+ * A spsc may be declared on a stack or statically and work as intended so long
```

Comment:
If you are anticipating your reader reading this as the letters S P S C then this should be "An SPSC". If on the other hand you expect them to do an inline replace as they are reading this to single-producer single-consumer, then "A" is the correct starting.  Either case, most acronyms are capatilized.